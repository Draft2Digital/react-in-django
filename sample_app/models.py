# coding=utf-8
from __future__ import absolute_import
from __future__ import division
from __future__ import print_function
from __future__ import unicode_literals

from django.db import models


########################################################################
class Author(models.Model):
    first_name = models.TextField()
    last_name = models.TextField()
    description = models.TextField()

    ####################################################################
    @property
    def display_name(self):
        return "{} {}".format(self.first_name, self.last_name)


########################################################################
class Book(models.Model):
    BookStatus = (
        ('DRAFT', 'Draft'),
        ('REQUESTED', 'Requested Listing'),
        ('PUBLISHED', 'Published'),
        ('DELISTED', 'Delisted'),
    )
    title = models.TextField()
    author = models.ForeignKey(Author)
    price = models.DecimalField(max_digits=5, decimal_places=2)
    status = models.CharField(max_length=30, choices=BookStatus)
