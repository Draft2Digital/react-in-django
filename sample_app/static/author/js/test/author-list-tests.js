"use strict";

import React from 'react';
import {mount, shallow} from 'enzyme';
import {shallowToJson} from 'enzyme-to-json';

import {expect} from 'chai';

import AuthorList from "../components/author-list.jsx";
import {AuthorStore} from '../models.js';
import AuthorItem from "../components/author-item";

//**********************************************************************
beforeEach(() => {
  let fixture = [
    {id: 1, first_name: "Amy", last_name: "Barnes", description: "one"},
    {id: 2, first_name: "Betty", last_name: "Cates", description: "two"},
    {id: 3, first_name: "Cathy", last_name: "Dean", description: "three"},
  ];
  AuthorStore.reset(fixture);
  expect(AuthorStore.expanded).to.equal(null);
});

//**********************************************************************
function getComponent() {
  return (
    <AuthorList authors={AuthorStore}/>
  );
}

//**********************************************************************
describe('AuthorList', () => {
  //******************************************************************
  it('wraps the AuthorItems', () => {
    let component = getComponent();
    let wrapper = shallow(component);
    expect(wrapper).to.exist;
    expect(wrapper.type()).to.equal('div');
    let ul = wrapper.childAt(0);
    expect(ul.type()).to.equal('ul');
    expect(ul.hasClass('authors')).to.equal(true);
    expect(ul.hasClass('collection')).to.equal(true);
    expect(ul.children().length).to.equal(3);  // 3 AuthorItems included
    let x = ul.childAt(0);
    expect(x.type()).to.equal(AuthorItem);

    /*
    let tree = shallowToJson(wrapper);
    expect(tree).toMatchSnapshot();
    */
  });
});
