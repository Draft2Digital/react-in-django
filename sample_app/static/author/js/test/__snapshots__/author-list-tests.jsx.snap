exports[`AuthorList author item changes when expanded 1`] = `
<div>
  <ul
    className="authors collection">
    <AuthorItem
      author={
        Object {
          "description": "one",
          "first_name": "Amy",
          "id": 1,
          "last_name": "Barnes",
        }
      }
      expanded={false} />
    <AuthorItem
      author={
        Object {
          "description": "two",
          "first_name": "Betty",
          "id": 2,
          "last_name": "Cates",
        }
      }
      expanded={false} />
    <AuthorItem
      author={
        Object {
          "description": "three",
          "first_name": "Cathy",
          "id": 3,
          "last_name": "Dean",
        }
      }
      expanded={false} />
  </ul>
  <a
    className="waves-effect btn"
    onClick={[Function]}>
    Refresh
  </a>
</div>
`;

exports[`AuthorList author item changes when expanded 2`] = `
<div>
  <ul
    className="authors collection">
    <AuthorItem
      author={
        Object {
          "description": "one",
          "first_name": "Amy",
          "id": 1,
          "last_name": "Barnes",
        }
      }
      expanded={false} />
    <AuthorItem
      author={
        Object {
          "description": "two",
          "first_name": "Betty",
          "id": 2,
          "last_name": "Cates",
        }
      }
      expanded={true} />
    <AuthorItem
      author={
        Object {
          "description": "three",
          "first_name": "Cathy",
          "id": 3,
          "last_name": "Dean",
        }
      }
      expanded={false} />
  </ul>
  <a
    className="waves-effect btn"
    onClick={[Function]}>
    Refresh
  </a>
</div>
`;
